trio/_core/tests/test_watchdog.py

```diff
@@ -0,0 +1,30 @@
+import threading
+import time
+from io import StringIO
+
+import contextlib
+
+from ..tests.tutil import slow
+from .. import _run
+
+
+@slow
+async def test_watchdog():
+    watchdog = _run.GLOBAL_RUN_CONTEXT.watchdog
+    target = StringIO()
+    with contextlib.redirect_stderr(target):
+        time.sleep(7)
+
+    assert target.getvalue().startswith(
+        "Trio Watchdog has not received any "
+        "notifications in 5 seconds, main "
+        "thread is blocked!"
+    )
```

Comment:
Maybe assert that `time.sleep` and `test_watchdog` show up in the output too, to make sure that the watchdog is printing tracebacks?